packages/workflow-designer/src/react/workflow-designer-context.tsx

```diff
@@ -206,6 +207,55 @@ export function WorkflowDesignerProvider({
 		[client, data.id, setAndSaveWorkspace],
 	);
 
+	const handleTriggerNodeCopy = useCallback(
+		async (sourceNode: Node, newNode: Node): Promise<void> => {
+			if (
+				!isTriggerNode(sourceNode) ||
+				!isTriggerNode(newNode) ||
+				sourceNode.content.state.status !== "configured"
+			) {
+				return;
+			}
+
+			try {
+				const originalTriggerId = sourceNode.content.state.flowTriggerId;
+
+				const originalTriggerResult = await client.getTrigger({
+					flowTriggerId: originalTriggerId,
+				});
+
+				if (originalTriggerResult?.trigger) {
+					const originalTrigger = originalTriggerResult.trigger;
+					const result = await client.configureTrigger({
+						trigger: {
+							workspaceId: originalTrigger.workspaceId,
+							nodeId: newNode.id,
+							enable: originalTrigger.enable,
+							configuration: originalTrigger.configuration,
+						},
+					});
+
+					if (result?.triggerId) {
+						workflowDesignerRef.current.updateNodeData(newNode, {
+							content: {
+								...newNode.content,
+								state: {
+									status: "configured",
+									flowTriggerId: result.triggerId,
+								},
+							},
+						} as Partial<TriggerNode>);
```

Comment:
You can remove type assertion:

```suggestion
						workflowDesignerRef.current.updateNodeData(newNode, {
							content: {
								...newNode.content,
								provider: "github",
								state: {
									status: "configured",
									flowTriggerId: result.triggerId,
								},
							},
						});
```

Reply:
Ah, you're right. I'm sorry. I think the original implementation is fine as is.